bundles/http/Citrix/SAP-Ariba-Demo/SAP Ariba HTTP Demo.js

```diff
@@ -0,0 +1,488 @@
+      CurrencyCode: 'USD', 
+      Quantity: 12,
+      ReportID: '12372847',
+      Total: 1446
+      },
+      {
+      ID: '19710704432971777', 
+      Supplier: 'Office Depot', 
+      Category: 'Office Supplies',
+      Description: 'General Stationary',
+      TransactionDate: date, 
+      Price: 30, 
+      CurrencyCode: 'USD', 
+      Quantity: 5,
+      ReportID: '12372847',
+      Total: 150
+      },
+      {
+      ID: '19710704432971778', 
+      Supplier: 'Microsoft', 
+      Category: 'Office Software',
+      Description: 'Microsoft 365 Subscription',
+      TransactionDate: date, 
+      Price: 30, 
+      CurrencyCode: 'USD', 
+      Quantity: 5,
+      ReportID: '12372848',
+      Total: 150
+      },
+      {
+      ID: '19710704432971779', 
+      Supplier: 'Electrician', 
+      Category: 'Maintenance',
+      Description: 'Fix lights',
+      TransactionDate: date, 
+      Price: 75.60, 
+      CurrencyCode: 'USD', 
+      Quantity: 1,
+      ReportID: '12372849',
+      Total: 75.60
+      }
+]
+
+const categories = [
+  {
+    Category: 'Appliances',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/folder.462b3c7e85bf4fa6e7a50801cd400bd2.svg'
+  },
+  {
+    Category: 'Maintenance',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/incident.e71280c2c22b2abd3aba06971728c296.svg'
+  },
+  {
+    Category: 'Training',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/course.fa9aabf2ac03cfad305b04ed8db96430.svg'
+  },
+  {
+    Category: 'Other',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/feedback.c533ed2d280d39d9288b2fd741a18651.svg'
+  },
+  {
+    Category: 'Office Supplies',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/case.2bb73f1b3eff52a8134fd591447bdaa1.svg'
+  },
+  {
+    Category: 'Office Software',
+    ImageURL: 'https://iws-stage-global-cdn-endpoint.azureedge.net/microapps/assets/exported/report_update.2d453d1646665c2ede6de83f848a6566.svg'
+  }
+]
+
+async function syncReports ({ dataStore, integrationParameters }) {
+  console.log('Synching data...')
+
+  const saveReports = calculateReportsWithTotals(expenses, reports)
+
+  dataStore.save('reports', saveReports)
+  dataStore.save('expenses', expenses)
+  dataStore.save('categories', categories)
+  console.log("Synchronization complete")
+}
+
+function calculateTotalAmount(reportId, expenses) {
+  return expenses.filter((expense) => expense.ReportID === reportId).reduce((result, expense) => result + expense.Price*expense.Quantity, 0)
+}
+
+function calculateReportsWithTotals(expenses, reports) {
+  return reports.map(report => ({...report, TotalAmount: calculateTotalAmount(report.ID, expenses)}))
+}
+
+function getRandomReport(list){
+  return JSON.parse(JSON.stringify(list[Math.floor(Math.random() * (list.length))]))
```

Comment:
I see 3 occurrances of `JSON.parse(JSON.stringify(....)`, it could be refactored to dedicated function `cloneViaJson`